router/pkg/mcpserver/operation_manager.go

```diff
@@ -0,0 +1,113 @@
+package mcpserver
+
+import (
+	"fmt"
+
+	"github.com/wundergraph/cosmo/router/pkg/schemaloader"
+	"github.com/wundergraph/graphql-go-tools/v2/pkg/ast"
+	"go.uber.org/zap"
+)
+
+// OperationsManager handles the loading and preparation of GraphQL operations
+type OperationsManager struct {
+	schemaDoc        *ast.Document
+	operations       []schemaloader.Operation
+	logger           *zap.Logger
+	schemaCompiler   *SchemaCompiler
+	excludeMutations bool
+}
+
+// NewOperationsManager creates a new operations manager
+func NewOperationsManager(schemaDoc *ast.Document, logger *zap.Logger, excludeMutations bool) *OperationsManager {
+	if logger == nil {
+		logger = zap.NewNop()
+	}
+
+	return &OperationsManager{
+		schemaDoc:        schemaDoc,
+		logger:           logger,
+		schemaCompiler:   NewSchemaCompiler(logger),
+		excludeMutations: excludeMutations,
+	}
+}
+
+// GetSimpleOperationInfoList gets the list of operations for the list_graphql_operations tool
+func (om *OperationsManager) GetSimpleOperationInfoList() []SimpleOperationInfo {
+	operations := make([]SimpleOperationInfo, 0, len(om.operations))
+	for _, op := range om.operations {
+		hasSideEffects := op.OperationType == "mutation"
+
+		// Skip mutation operations if ExcludeMutations is enabled
+		if hasSideEffects && om.excludeMutations {
+			continue
+		}
+
+		operations = append(operations, SimpleOperationInfo{
+			Name:           op.Name,
+			Description:    op.Description,
+			OperationType:  op.OperationType,
+			HasSideEffects: hasSideEffects,
+		})
+	}
+
+	return operations
+}
+
+// LoadOperationsFromDirectory loads operations from a specified directory
+func (om *OperationsManager) LoadOperationsFromDirectory(operationsDir string) error {
```

Comment:
Should OperationsManager know about directories? Or should we abstract the "operations storage" into an interface and make OperationsManager independent of storage implementations?

Reply:
I like the idea but seems overkill for the first iteration. Once we connect cosmo to it we have a reason to refactor it.